Handle `halt` params of `Mimir`

veado opened the pull request:
- [x] `useMimirHalt` hook
- [x] Pool helpers `disableAllActions`, `disableTradingActions` (incl. tests)
- [x] `disableRuneUpgrade` helper (incl. tests)
- [x] [AssetTable] Disable `Upgrade` buttons if needed
- [x] [AssetDetails] Disable `Send`/`Upgrade`/`Deposit` buttons if needed
- [x] [AppView] Update alert handling to show disabling upgrade 
- [x] Override `halted` chains status received from `inbound_addresses` endpoint with data received from `Mimir`
- [x] Update stories

Closes #1645

veado committed:
Handle `halt` params of `Mimir`

gromxyz commented on src/renderer/helpers/assetHelper.test.ts:
```diff
@@ -281,4 +283,25 @@ describe('helpers/assetHelper', () => {
+    it('disabled for BNB.RUNE + halted THORChain', () => {
+      expect(disableRuneUpgrade({ asset: AssetRune67C, haltThorChain: true, haltEthChain: false })).toBeTruthy()
+    })
+    it('ebabled for BNB.RUNE + halted ETH chain', () => {
```
Just a small typo `ebabled` => `enabled `

gromxyz commented on src/renderer/helpers/assetHelper.test.ts:
```diff
@@ -281,4 +283,25 @@ describe('helpers/assetHelper', () => {
+    it('ebabled for BNB.RUNE + halted ETH chain', () => {
+      expect(disableRuneUpgrade({ asset: AssetRune67C, haltThorChain: false, haltEthChain: true })).toBeFalsy()
+    })
+    it('ebabled for BNB.RUNE + no halted chains', () => {
```
One more small typo

veado committed:
Override `halted` (inbound_addresses) by mimir

values everywhere. Update `halt` + `upgrade` warning alert handling in
AppView.

veado committed:
Fix build by extracting types + const

from useMimirHalt hook

- [AppView] Revert removing `rdAltOnPending` in `renderHaltedChainsWarning`

veado committed:
Remove currying, use use object args

veado committed:
Test `disableAllActions`

veado committed:
Test `disableTradingActions`

veado committed:
Consider `disableTradingActions` for swap

veado committed:
Merge branch 'develop'

into issue-1645/mimir-halt
